lib/elixir/lib/module/types/descr.ex

```diff
@@ -783,6 +844,10 @@ defmodule Module.Types.Descr do
         end)
       end)
     end)
+    |> case do
+      [] -> 0
```

Comment:
@gldubc you will have to do the same for tuples. We no longer match empty list as empty on intersection/difference. It is either zero or an empty map.